### ISSUE TITLE
Comment on this issue when beta is refreshed.

### ISSUE THREAD

**Issue by kdahlquist** [dondi/GRNsight#406]
I'm opening this issue so that you can post a comment here when the GRNsight beta server is refreshed, so that @kdahlquist (and others) will get an e-mail notification.

**Comment by NAnguiano** [comment]
The beta has been refreshed with the fix for issue #408. It also includes the tweaks from #397 and #395.

**Comment by NAnguiano** [comment]
Beta has been refreshed! It has issues #396 #397 #398 #399 and #419.

**Comment by NAnguiano** [comment]
For anyone who refreshes beta in the future, keep in mind that the node apps are now being run from the top directory, not the server and web-client directories. The apps are called using server/app.js and web-client/app.js. This is a direct result of the `npm start` implementation. `npm start` can't be used on the beta server itself because of the additional requirements, so running the apps on the parent directory is now required.

**Comment by NAnguiano** [comment]
Beta was refreshed with the new pull request just before the meeting. It included the fixes for #395, 
#396, and #399.

**Comment by kdahlquist** [comment]
Did you increment the version number?  Shouldn't we be on 2.0.3 now?  I will look at it and post comments on the relevant issues.

**Comment by NAnguiano** [comment]
The version number is now properly incremented.

**Comment by NAnguiano** [comment]
Beta has been refreshed from PR #446 and #432.

**Comment by NAnguiano** [comment]
Beta has been refreshed from PR #448.

**Comment by NAnguiano** [comment]
Beta is refreshed from PR #459.

**Comment by kdahlquist** [comment]
Did Beta get refreshed from PR #460?  It doesn't seem so.

**Comment by eileenchoe** [comment]
My bad -- it did not. I need to review how to refresh beta, I haven't actually done a refresh myself yet.

**Comment by kdahlquist** [comment]
Since you @anuvarsh, @NAnguiano, and @mihirsamdarshi  seem to be working right now, will somebody refresh beta from PR #460 so it is up to date with the beta branch?  Thanks.

**Comment by NAnguiano** [comment]
@kdahlquist @eileenchoe I updated it about 10 mins ago, so it should be up to date now!

**Comment by kdahlquist** [comment]
Got it!

**Comment by NAnguiano** [comment]
@eileenchoe I made a quick guide on the wiki to refresh the server right [here](https://github.com/dondi/GRNsight/wiki/Quick-Guide-to-Refreshing-GRNsight). There's a more detailed guide on the wiki if you want to read it as well. :)

**Comment by NAnguiano** [comment]
beta has been refreshed

**Comment by dondi** [comment]
While the main `beta` line is in a pause as @NAnguiano finishes up, I figured it wouldn't hurt to upload the temporary `normalization` branch from @anuvarsh. This version, bumping to 2.0.17, includes the latest normalization implementation, including restoration of the previously lost functionality.

This version was branched _from_ `beta` so all of the features from that branch as of May the 4th (be with you) should still be available anyway.

**Comment by dondi** [comment]
Beta points back to the `beta` branch now so it is at 2.0.16. If this branch is updated and changed, we should skip to 2.0.18 because `normalization` was inserted in between.

**Comment by kdahlquist** [comment]
Just noting that beta has been refreshed by @NAnguiano to 2.0.18.

**Comment by NAnguiano** [comment]
Beta is refreshed from PR #458 :)

**Comment by eileenchoe** [comment]
Beta refreshed from PR #486, #490, and #492!

**Comment by eileenchoe** [comment]
@kdahlquist  Versioning question: just noticed that master is at v2.0.20, shouldn't it be v2.0.0?

**Comment by kdahlquist** [comment]
No, we decided that we would increment it from the last functioning beta.  So all betas and masters get unique version numbers.  It seemed odd to increment the middle number (like to 2.1.0) since we had never had an official release of 2.0.x, so we went with 2.0.20, which does have a nice ring to it :)

We should target having a master release by June 3 to update with all the stuff you guys are working on now.  At that point, we'll go to 2.1.x.

If we have to make any hot fixes on master, we would go to 2.0.21 on master, but that seems unlikely right now.

Make sense?

**Comment by eileenchoe** [comment]
Beta refreshed from PR #500

**Comment by eileenchoe** [comment]
Beta refreshed from PR #502. Beta is now at v2.1.4

**Comment by eileenchoe** [comment]
Beta refreshed from PR #504. Beta is now at v2.1.5

**Comment by eileenchoe** [comment]
Beta has been refreshed from PR #509. Beta is now at v2.1.6.

**Comment by mihirsamdarshi** [comment]
Beta has been refreshed from PR #511. Beta is now at v2.1.7

**Comment by kdahlquist** [comment]
Beta is refreshed from PR #516.  Beta is now at v2.1.8

**Comment by mihirsamdarshi** [comment]
Beta has now been refreshed from PR #517. Beta is now at v2.1.10

**Comment by dondi** [comment]
I wasn't quite sure where this should go but I figured this would work since future betas may well tweak our sliders 🙃

https://www.designernews.co/stories/84443-redditors-design-worst-volume-sliders-possible

**Comment by kdahlquist** [comment]
@dondi, you pushed changes to the beta server as part of issue #513.  However, you didn't comment on this issue.  The date on the beta page is still from June instead of September and I don't know if the version number was incremented.  Will you please confirm that what it says there is correct or fix?  Thanks.

**Comment by dondi** [comment]
Ah yes, sorry, this part of the workflow had fallen out of my radar. We are now at v2.2.1 and the beta page has been updated accordingly, along with the date. The new test files from #515 and #513 have also been committed to the `beta` branch, under the _test-files/edge-tests_ folder.

@eileenchoe’s devops work (PR #528) and @ebachoura’s linting fix (PR #525) are also part of this version, although these modifications (plus the new test files) do not directly affect the user-visible web app at http://dondi.github.io/GRNsight/beta.

**Comment by eileenchoe** [comment]
GRNsight (beta & master) is down and I can't ssh into the server. @ebachoura says that apparently the biodb website is also down, so it could be a similar issue.

**Comment by kdahlquist** [comment]
Hmm.  Both sites are up right now.  Maybe a web connectivity issue?

**Comment by eileenchoe** [comment]
Beta is refreshed from @yshin4 's PR #529. Now at v 2.2.2.

**Comment by eileenchoe** [comment]
Interesting, @kdahlquist none of us could access it last night but it does seem fine now!

**Comment by dondi** [comment]
v2.2.3 is online. The visible change is @yshin4’s proposed fix to #188. The `beta` branch also now has her added test files, which she placed in _test-files/edge-tests_.

**Comment by yshin4** [comment]
Updated to v2.2.4. This version contains the fixes to #392 #318 #188.

**Comment by yshin4** [comment]
Updated to v2.2.5. This version properly loads demo files.

**Comment by eileenchoe** [comment]
Beta refreshed. Now at version 2.2.9 with PR #552

**Comment by eileenchoe** [comment]
Refreshed beta. Now at v2.3.1 with PR #565

**Comment by yshin4** [comment]
Refreshed beta to v.2.3.2 with PR #569

**Comment by dondi** [comment]
Back up and running.

**Comment by eileenchoe** [comment]
Refreshed beta to v.2.3.3 with PR #567

**Comment by yshin4** [comment]
Refreshed beta to v.2.3.4 PR #572

**Comment by mihirsamdarshi** [comment]
beta refreshed to v.2.3.5 PR #580

**Comment by kdahlquist** [comment]
Don't forget to update the last modified date.  I fixed it just now.

**Comment by eileenchoe** [comment]
beta refreshed to v.2.3.6 with PR #582, node coloring is now live! 😀

**Comment by yshin4** [comment]
beta refreshed to v.3.0.0 with master 3.0.0 release.

**Comment by mihirsamdarshi** [comment]
Beta updated to v.3.0.2 with PR #595

**Comment by dondi** [comment]
Minor bump to v3.0.4 after some necessary hot fixes were discovered and made.

@mihirsamdarshi and @johnllopez616 you will want to merge `beta` back into your respective branches.

**Comment by johnllopez616** [comment]
Beta has been refreshed to v3.0.6 following PR #623

**Comment by johnllopez616** [comment]
Beta has been refreshed to v3.0.7 following PR #624

**Comment by kdahlquist** [comment]
I'm getting a 502: bad gateway error when I try to load the beta page.

**Comment by kdahlquist** [comment]
OK, I can see beta.  Maybe I caught it in the middle of the refresh?

**Comment by johnllopez616** [comment]
Beta has been refreshed to v 3.0.8 following PR #626

**Comment by mihirsamdarshi** [comment]
Beta has been refreshed to v3.0.9 following PR #628

**Comment by johnllopez616** [comment]
Beta has been refreshed to v3.0.11 following PR #636

**Comment by johnllopez616** [comment]
Beta refreshed to v. 3.0.12 following PR #637

**Comment by johnllopez616** [comment]
Refreshed to v 3.0.15 following PR #656 and #657

**Comment by kdahlquist** [comment]
Beta was refreshed to v3.0.16 to preview changes for 3.1.0 release.

**Comment by johnllopez616** [comment]
Beta refreshed to v 3.1.4 following PR #677

**Comment by johnllopez616** [comment]
Beta refreshed to v 3.1.5 following PR #686

**Comment by mihirsamdarshi** [comment]
Beta refreshed to v3.1.6 following PR #685

**Comment by mihirsamdarshi** [comment]
beta refreshed to v3.1.7 following PR #693

**Comment by kdahlquist** [comment]
Beta is refreshed to 3.1.8.

**Comment by kdahlquist** [comment]
Please update beta with PR #705.  After, don't forget to update the date and version on gh-pages.

**Comment by afiller1** [comment]
Beta is refreshed to v3.1.10, 2/12/2019.

**Comment by dondi** [comment]
Beta has been updated to v3.1.11, revised 2/12/2019. The changes from v3.1.10 include a new "clean-room" package-lock.json plus some late-breaking bug fixes in the MVC implementation. The Export to Excel Workbook feature is now also available from the File menu, under Export Data.

**Comment by mihirsamdarshi** [comment]
Beta was updated to v3.1.12 on 2/24/2019 after PR #731

**Comment by mihirsamdarshi** [comment]
Beta was updated to v3.1.13 on 3/26/2019 after PR #739

**Comment by mihirsamdarshi** [comment]
Beta was updated to v3.1.15 on 4/30/2019 after PR's #738 and #743

**Comment by mihirsamdarshi** [comment]
Beta has been updated to v3.1.17 on 5/28/2019 following PR's #749 & #756

**Comment by mihirsamdarshi** [comment]
Beta has been update to v3.1.21 on 6/5/2019 following PR's #757, #761, #762 & #764

**Comment by mihirsamdarshi** [comment]
Beta has been updated to v3.1.22 on 6/6/2019 following PR #771

**Comment by kdahlquist** [comment]
Don't forget to increment the version and date :)

**Comment by mihirsamdarshi** [comment]
Beta has been updated to v3.1.23 on 6/7/2019 following PR #780

**Comment by mihirsamdarshi** [comment]
Beta has been updated to v3.1.24 on 6/7/2019 following PR #783

**Comment by dondi** [comment]
@mihirsamdarshi I’m still seeing the https/http error for `/server/ga?path=_____` URLs (see screenshot). Please review to see if something was missed in this update, thanks!

(note that I purposely didn’t allow Chrome to "load insecure scripts" because our http/https fixes should not make this necessary…I thought this had been addressed previously so this might be a regression)

<img width="1110" alt="image" src="https://user-images.githubusercontent.com/382242/59132463-1d9d7180-892a-11e9-98fb-56e680189b88.png">

**Comment by mihirsamdarshi** [comment]
Beta has been refreshed to v4.0.0 following PR #786 on 6/8/2019

**Comment by dondi** [comment]
Beta v4.0.1 is now online, seeking to address issues #781, #777, #773, #770, #750, and #615.

**Comment by dondi** [comment]
Beta v4.0.2 is now online, with updates to issues #773, #781, and #792. Issue #59 is also addressed partially, with PNG export now available (but not the other formats). Also, there are comments in #791 that need further investigation/confirmation.

**Comment by dondi** [comment]
Beta v4.0.3 is now online, with updates to issues #773 and #794. Pull request #796 could not be merged because Travis reports a failing test on it.

**Comment by dondi** [comment]
Beta v4.0.4 is now online, with fixes for issues that @mihirsamdarshi was working on.

**Comment by mihirsamdarshi** [comment]
Beta has been refreshed to v4.0.5 following PR #802

**Comment by mihirsamdarshi** [comment]
Beta has been refreshed to v4.0.6 following PR #805

**Comment by mihirsamdarshi** [comment]
Beta has been refreshed to v4.0.7 following PR #807

**Comment by mihirsamdarshi** [comment]
Beta has been refreshed to v4.0.8 following PRs #806 and #812

**Comment by dondi** [comment]
Beta has been updated to v4.0.9, reflecting assorted visual/layout cleanup relating to #803.

**Comment by dondi** [comment]
Beta v4.0.10 is out, now fixing #817 so that #803 can continue for the Node section.

**Comment by dondi** [comment]
Beta has been updated to v4.0.11, with #803 Node section visual issues now addressed.

**Comment by dondi** [comment]
Beta has been updated to v4.0.12, integrating #841 initial user interface changes toward species selection. @Kevboe please feel free to provide a description of what changed and what to look for in terms of what you’ve added; thanks!

**Comment by kdahlquist** [comment]
@afiller1 and @Kevboe, Are there going to be any pull requests and new versions of beta before the meeting this Monday?  I would like to be able to review and offer feedback during the meeting.

**Comment by dondi** [comment]
Beta has been updated to v4.0.13, integrating #843 latest user interface changes toward species selection. Issues found during the 2020-04-06 meeting are addressed here; the next step is to sync up the API code, some of which now appears to be working differently or returning different results from before. @Kevboe this is in _api.js_—that’s where to look next, assuming that the gene page URL and its processing now look right to you.

**Comment by dondi** [comment]
Beta has been updated to v4.0.14, integrating #844 from @Kevboe. I will seek to update it again if @afiller1 is able to address the Travis issues for #845. But for now at least this has @Kevboe’s updates for this week.

**Comment by dondi** [comment]
Beta has been updated to v4.0.15, integrating #846 API fixes from @Kevboe.

The changes were also accidentally placed in `master` and that will be difficult to undo, but that will remain unreleased anyway so we can just let it be.

**Comment by dondi** [comment]
Beta has been updated to v4.0.15, integrating #845 from @afiller1 and #847 from @Kevboe.

Database configuration needs to be done (perhaps this can happen at our next meeting) so although the code is there, settings need to be established before expression database functionality can be observed.

**Comment by dondi** [comment]
Beta has been updated to v4.0.17 with #850 and #851 being the primary additions.

@afiller1 and @Kevboe now that we have merges of your code, please take a look when you get a chance to ensure that the changes that you expect are indeed manifested here. Thanks!

**Comment by dondi** [comment]
Beta has been updated to v4.0.18, merging #859, #864, and #866.

I imagine that this will be _the_ version that will be used for the senior project presentations so @afiller1 and @Kevboe please put it through its paces so that it demoes correctly.

If you discover any need for hot fixes, go ahead and commit+push to beta directly, and pull from the beta server to update it.

**Comment by dondi** [comment]
Beta has been updated to v4.0.19, merging #868 and #876. #876 fixes the gray threshold bug #832.

**Comment by dondi** [comment]
During the 9/18 meeting it was discovered that the grnsight.cs.lmu.edu server had run out of disk space. A temporary measure was found by deleting certain files that the beta didn't need but @dondi will look further into this offline to see either where additional space can be reclaimed or if unavoidable, the EC2 host might need to get more disk space.

**Comment by dondi** [comment]
Beta has been updated to v4.0.20, merging #879 from @Onariaginosa, consisting of cumulative work done so far with testing expression data sheets. The beta has also been updated to use v4.2.2 of [multiparty](https://www.npmjs.com/package/multiparty) for compatibility with newer versions of node.

**Comment by dondi** [comment]
Beta has been updated to v4.0.21, merging #880 from @igreen1 to fix #444 and #700.

**Comment by dondi** [comment]
Beta has been updated to v40.022, merging #881 from @luluriffic to fix #831.

**Comment by dondi** [comment]
Beta has been updated to v4.0.23, merging #885 from @igreen1 with additional tweaks relating to #444, this time specific to self-referring edges.

**Comment by dondi** [comment]
Beta has been updated to v4.0.24, merging #887 from @luluriffic, which addresses part of #804.

**Comment by dondi** [comment]
Beta has been updated to v4.025, merging #895 from @Onariaginosa with refactored Warnings code (#892) and new adjacency matrix data tests (#894).

**Comment by dondi** [comment]
Beta has been updated to v4.0.26, merging #896 from @igreen1. This addresses issue #886 (handle missing meta in GraphML and SIF) and #444 (margin of the self-referential edges).

**Comment by dondi** [comment]
Beta has been updated to v4.0.27, merging #898 from @Onariaginosa which consists of a full fix for #417 (missing value error checking & testing, plus some refactoring).

**Comment by dondi** [comment]
Beta has been updated to v4.0.28, merging #900 from @Onariaginosa and #901 from @igreen1 (#900 has actually been on beta for the past week but I didn’t get around to updating the version until now, and #901 was already out there in the wild so I ended up merging that one also).

**Comment by dondi** [comment]
Beta has been updated to v4.0.29, merging #902 from @Onariaginosa.

**Comment by dondi** [comment]
Beta has been updated to v4.0.30, merging #909 from @igreen1 and #910 from @mihirsamdarshi.

**Comment by dondi** [comment]
Beta has been updated to v4.0.31, merging #915 from @Onariaginosa for issues #545, #570. Thanks also to @igreen1 for walking through the PR checklist for the first time in a while! 💯

**Comment by dondi** [comment]
Beta has been updated to v4.0.32, merging PR #918 from @Onariaginosa for issue #917. PR checklist was fully checked before merging—thanks @igreen1 and @ahmad00m!

**Comment by dondi** [comment]
Beta has been updated to v4.0.33, merging PR #920 from @Onariaginosa for Excel export issues #919, #907. PR checklist done by @dondi

**Comment by dondi** [comment]
Beta has been updated to v4.0.34, merging PR #921 and #922 to address the fixes listed there. A couple of PR checklist item issues found by @ahmad00m but we can seek to resolve those individually in beta.

**Comment by dondi** [comment]
Beta has been updated to v4.0.35, merging PR #926.

**Comment by dondi** [comment]
Beta has been updated to v4.0.36—this version seeks to close out the last functionality loose end from the end of the semester involving viewport resizing.

I realize it’s summer and no one is on the hook for anything, but if anyone has a free moment to check this out and provide feedback, that would be much appreciated 🤗 Thanks!

**Comment by dondi** [comment]
The server update has been completed and GRNsight execution (both production and beta) has been restored. The beta page now shows v5.0.0 with today’s date.

I have also applied the new environment variables for connecting to the expression database on production, and in my testing that part now works.

The version of node on this latest Ubuntu server is indeed newer than the prior one, but still not the newest version (Ubuntu Long-Term-Service releases tend to skew very conservative in their version updates). I will see whether this new version is sufficient for the planned security updates. Stay tuned.

Final server uptime before reboot is as follows:
![image](https://user-images.githubusercontent.com/382242/133909939-26eec230-56e2-4c63-a232-a7de71a8f2ff.png)

**Comment by dondi** [comment]
Following the merge of #985 the deployed beta is now at 6.0.2

A quick check seems to show a regression in database functionality though…genes that are appearing on prod do not seem to appear on beta now (for me). Maybe we can look at it at the meeting

**Comment by dondi** [comment]
Epic PR #996 has been merged into `beta`, which is now at 6.0.3

I have verified and closed the 6.0.2 regression bug #995 but have not had a chance to look into the other bugs on `beta`. If the team can please run `beta` through its paces in the next few days, we can close the remaining `review requested` items and potentially merge this into production

@Onariaginosa I think you had some follow-up commits relating to database scripting and documentation—all clear to do this

@Sarronnn you can now queue up your PR for #993 after merging `beta` into your branch (or maybe even starting a fresh branch from `beta`) and restoring the refactor work that you have done

**Comment by dondi** [comment]
PR #1025 has been merged into `beta`, bringing it to v6.0.6

**Comment by ahmad00m** [comment]
PR #1026 have been merged into beta. The version and modified date has been changed to reflect the new changes. beta is currently at v6.0.7

**Comment by Onariaginosa** [comment]
PR #1039 and #1040 has been merged into `beta`. The version and modified date has been updated. `beta` is now at v7.0.0

**Comment by dondi** [comment]
PR #1045, containing the fix for #1016, has been merged into `beta`. Version has been updated to v7.0.1

**Comment by dondi** [comment]
`beta` now has PR #1048 and PR #1049, holding fixes to #1033 and #1034

**Comment by dondi** [comment]
`beta` is now at 7.0.5, finally including PR #1037, which addresses #992. Local databases merging `beta` from this point will need to be revised accordingly

**Comment by dondi** [comment]
`beta` has been updated to 7.0.6 integrating PR #1063 to address issue #1050. Some notes have already been made on refining this fix further

**Comment by dondi** [comment]
`beta` is now on 7.0.7 integrating PR #1064 with refinements for issue #1050

**Comment by dondi** [comment]
`beta` is now on 7.0.8 integrating PRs #1073 and #1075 addressing issues #1072 and #1065 respectively

**Comment by dondi** [comment]
`beta` is now on 7.0.9 integrating @ceciliazaragoza’s code style changes address issue #1002

**Comment by akaiap** [comment]
`beta` is now on 7.0.12 integrating @ntran18 #1094

**Comment by ntran18** [comment]
`beta` is now on 7.0.13 integrating @ntran18 #1096 and #1101 to address issue #1079, #1097 to address issue #1054, and #1100 to fix issue #1099

**Comment by ceciliazaragoza** [comment]
`beta` is now on 7.0.14 integrating @ceciliazaragoza integrating PR #1109 to address issue #1081 and @ntran18 integrating PR #1105 to address issue #1089.

**Comment by ceciliazaragoza** [comment]
`beta` is now on 7.0.15 integrating @ceciliazaragoza integrating PR #1112 and PR #1114 to address issue #1083 .

**Comment by dondi** [comment]
`beta` is now on 7.1.4 integrating PRs #1123, #1124, and #1125

**Comment by dondi** [comment]
`beta` has been updated to 7.1.5 with PRS #1127 and #1128

**Comment by dondi** [comment]
It took a little more time due to some delays on my part and additional review iterations, but `beta` is now at 7.1.6! PRs in this version include: #1129, #1131, #1132, #1133, #1134, #1136, and #1137. These PRs address issues #1080, #1130, #1059, #1093, #1098, #1068, #1107, #1108

That’s quite a few PRs and issues addressed—it’s truly bug-bash time 🐜🔨

**Comment by dondi** [comment]
We have add a new `code review` tag to distinguish issues that are deployed and need QA/verification vs. issues that are on PRs. @akaiap @ceciliazaragoza @dondi @kdahlquist will seek to verify the remaining `review requested` issues on beta with an eye toward closing them out next week

**Comment by dondi** [comment]
`beta` has been updated to 7.1.7, including candidate fixes for #1053, #1093, and #59

**Comment by dondi** [comment]
`beta` is now at 7.1.8, now including GitHub Actions continuous integration (CI) via #1118 and a fix for #1142

**Comment by dondi** [comment]
`beta` is now at 7.1.9. with the main change being a renaming of our environment variables to purely `DB_USERNAME` and `DB_PASSWORD`, reflecting that our database now contains more than just expression data (#1030)

Folks with local copies should, after merging this latest `beta` version, rename their local environment variables accordingly (i.e., no more `EXPRESSION_` in the beginning)

**Comment by dondi** [comment]
`beta` has been updated to 7.1.10, with the hallmark of this release being focused on our transition to GitHub Actions: #1145, #1146, and #1147

**Comment by dondi** [comment]
`beta` has been updated to 7.1.11 to include the fix for #1144

**Comment by dondi** [comment]
`beta` has been updated to 7.2.2 to include for #1082

**Comment by dondi** [comment]
`beta` has been updated to 7.2.3 to include the tickets below, but it has stopped functioning. Upon looking at the server logs, it appears that the age of Node on our server (12.22.9) has caught up with us; it does not recognize the `export` statement which is in our latest code

I will look into what options we have for adjusting to this

Meanwhile, the tickets are: #863, #1150, #1173, and our big AllianceMine/database migration (#1120, #1164, #1165). For the latter, this is a code change only; the actual migration should happen first on local setups, and then when those are validated, we can do this on the server

**Comment by dondi** [comment]
`beta` appears to be back up and running now. Shoutout @ceciliazaragoza for the hotfix…worked on and pushed while @ntran18 was presenting her work during the Interdisciplinary Research Symposium 😁

I retained the version and revision date since it’s best to view this as a hotfix

**Comment by dondi** [comment]
Additional hotfixes have been applied, but they currently assume that the new namespace exists and so an attempt to Load from Database generates an error. Next steps will be to create this namespace in our database server

**Comment by dondi** [comment]
Beta has been updated v7.2.5 containing PRs merged during the 4/23 meeting

**Comment by dondi** [comment]
Beta has been updated to v7.2.7, containing PR #1225 in order to address lingering non-sorting behavior from #1191

**Comment by dondi** [comment]
Beta has been updated to v7.3.0, notably containing PR #1234 which addresses issue #1227, renaming the _web-client_ folder to _web-client-classic_

The release also includes:
* PR #1245 for issue #1211 
* PR #1246 for issue #1208 
* PR #1246 for issue #1214

**Comment by dondi** [comment]
Beta has been updated to v7.3.1, mainly with changes made to allow for concurrent support of both the classic and React versions of the app, particularly with CORS

And as it turns out, as of now those adjustments appear to have regressed our deployed version: @ceciliazaragoza database-related requests are now producing 500 errors. Internally, the stack traces are as follows. It appears that changes made to support CORS for the React version are preventing CORS in the deployed version:
```
Error: Request not allowed by CORS
    at origin (/data/GRNsight-beta/server/app.js:36:42)
    at /data/GRNsight-beta/node_modules/cors/lib/index.js:223:13
    at optionsCallback (/data/GRNsight-beta/node_modules/cors/lib/index.js:204:9)
    at corsMiddleware (/data/GRNsight-beta/node_modules/cors/lib/index.js:209:7)
    at Layer.handle [as handle_request] (/data/GRNsight-beta/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/data/GRNsight-beta/node_modules/express/lib/router/index.js:317:13)
    at /data/GRNsight-beta/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/data/GRNsight-beta/node_modules/express/lib/router/index.js:335:12)
    at next (/data/GRNsight-beta/node_modules/express/lib/router/index.js:275:10)
    at cors (/data/GRNsight-beta/node_modules/cors/lib/index.js:179:7)
    at /data/GRNsight-beta/node_modules/cors/lib/index.js:229:17
    at originCallback (/data/GRNsight-beta/node_modules/cors/lib/index.js:218:15)
    at /data/GRNsight-beta/node_modules/cors/lib/index.js:223:13
    at optionsCallback (/data/GRNsight-beta/node_modules/cors/lib/index.js:204:9)
    at corsMiddleware (/data/GRNsight-beta/node_modules/cors/lib/index.js:209:7)
    at Layer.handle [as handle_request] (/data/GRNsight-beta/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/data/GRNsight-beta/node_modules/express/lib/router/index.js:317:13)
    at /data/GRNsight-beta/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/data/GRNsight-beta/node_modules/express/lib/router/index.js:335:12)
    at next (/data/GRNsight-beta/node_modules/express/lib/router/index.js:275:10)
    at methodOverride (/data/GRNsight-beta/node_modules/method-override/index.js:65:14)
    at Layer.handle [as handle_request] (/data/GRNsight-beta/node_modules/express/lib/router/layer.js:95:5)
GET /grnsettingsdb?type=DefaultDataset 500 1.518 ms - 2281
Error: Request not allowed by CORS
    at origin (/data/GRNsight-beta/server/app.js:36:42)
    at /data/GRNsight-beta/node_modules/cors/lib/index.js:223:13
    at optionsCallback (/data/GRNsight-beta/node_modules/cors/lib/index.js:204:9)
    at corsMiddleware (/data/GRNsight-beta/node_modules/cors/lib/index.js:209:7)
    at Layer.handle [as handle_request] (/data/GRNsight-beta/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/data/GRNsight-beta/node_modules/express/lib/router/index.js:317:13)
    at /data/GRNsight-beta/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/data/GRNsight-beta/node_modules/express/lib/router/index.js:335:12)
    at next (/data/GRNsight-beta/node_modules/express/lib/router/index.js:275:10)
    at cors (/data/GRNsight-beta/node_modules/cors/lib/index.js:179:7)
    at /data/GRNsight-beta/node_modules/cors/lib/index.js:229:17
    at originCallback (/data/GRNsight-beta/node_modules/cors/lib/index.js:218:15)
    at /data/GRNsight-beta/node_modules/cors/lib/index.js:223:13
    at optionsCallback (/data/GRNsight-beta/node_modules/cors/lib/index.js:204:9)
    at corsMiddleware (/data/GRNsight-beta/node_modules/cors/lib/index.js:209:7)
    at Layer.handle [as handle_request] (/data/GRNsight-beta/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/data/GRNsight-beta/node_modules/express/lib/router/index.js:317:13)
    at /data/GRNsight-beta/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/data/GRNsight-beta/node_modules/express/lib/router/index.js:335:12)
    at next (/data/GRNsight-beta/node_modules/express/lib/router/index.js:275:10)
    at methodOverride (/data/GRNsight-beta/node_modules/method-override/index.js:65:14)
    at Layer.handle [as handle_request] (/data/GRNsight-beta/node_modules/express/lib/router/layer.js:95:5)
```

In order to facilitate triage and repair of this error, I’m keeping this broken version online for now. I can redeploy when a fix is found, or also rollback if it turns out to be a blocker for other work. But for now I think it helps to be able to examine this behavior live because it does not happen on a locally-running copy

**Comment by dondi** [comment]
Beta has been updated to v7.3.3, integrating @MilkaZek’s PRs that were merged on 11/5

**Comment by dondi** [comment]
Some merges have been done to `beta` but they have not yet been deployed because there are potentially a few more PRs to integrate. Depending on the timing, I’ll either deploy eventually or deploy with the other pending PRs included

**Comment by dondi** [comment]
Beta has been updated to v7.5.8 with for #1296, #1370, and #1275 

This can be considered as the release candidate, to become v7.6.0 unless any late-breaking issues are found

**Comment by dondi** [comment]
Beta has been updated to v7.5.9 integrating #1368 and #1378, both of which address #1192. #1378 also appears to address the regression to #1323 but that may require further verification

**Comment by dondi** [comment]
Beta has been updated to 7.5.11, including the fix for #1363

**Comment by dondi** [comment]
Beta has been updated to 7.7.0, reflecting the production release 7.6.0. It already has a new potential bug fix, for #1397

**Comment by kdahlquist** [comment]
@dondi did you pull main into beta?  I made a hotfix on the demo files on main so that our documentation is now in alignment with our standardization of cell A1 of the network sheet.

**Comment by dondi** [comment]
I did merge `main` into `beta`, yes…I’d say around a little before 2pm? Was the hotfix on `main` by then?

**Comment by kdahlquist** [comment]
Yes, I see the change on beta so we are all good!

**Comment by dondi** [comment]
Beta has been updated to 7.7.1 with an additional potential bug fix, for #1364

**Comment by dondi** [comment]
Beta has been updated to 7.7.2, containing potential bug fixes for #1397 (the missing period), #1388, and #1273

**Comment by dondi** [comment]
Beta has been updated to 7.7.3, containing potential bug fixes for #1425, the updated warning text for #1404, and the updated AllianceMine URL (inline for now, ideally to be made configurable later). Because #1425 has likely been addressed, testing on #1273 can also resume as well

**Comment by dondi** [comment]
Beta has been updated to 7.7.4, with potential bug fixes for: #1339, #1355, #1395, #1399, #1439, #1442. On the React side, the `beta` now also includes #1411, although this does not affect the classic deployment

**Comment by dondi** [comment]
Beta has been updated to 7.7.5 now including the big test refactor PR from @ntran18 which potentially addresses #1394, #1398, and overall #1422

**Comment by dondi** [comment]
Beta has been updated to 7.7.6, including fixes for #1229, #1339, #1384, #1418, and #1456

**Comment by dondi** [comment]
Beta has been updated to 7.7.7, including fixes for #1557, #1558,  #1559, #1560, and #1357

**Comment by dondi** [comment]
Beta has been updated to 7.7.8, including the fix for #1375

**Comment by dondi** [comment]
Beta has been updated to 7.7.9, including fixes for #1357, #1387, #1428, #1443, and #1449

**Comment by dondi** [comment]
Beta has been updated to 7.9.1, including fixes for #1384, #1561  and #1587

**Comment by dondi** [comment]
Beta has been updated to 7.9.2, including fixes for #1561 and #1557

**Comment by dondi** [comment]
Beta has been updated to 7.9.3, including updating wording for the message in #1561

**Comment by dondi** [comment]
Beta has been updated to 7.9.4, including new tests and less redundant code for #1557

**Comment by dondi** [comment]
Beta has been updated to 7.9.5, including the non-user-facing #1606 fix (verified via test suite execution) and #1387 

Although CI appears to have failed for the recent `beta` merges, it was determined that this was due to the Coveralls website being offline at the time. The codebase itself built and tested successfully, so the failures did not block deployment

**Comment by dondi** [comment]
We’re on the verge of a new beta release but I’d like to wait for a requested revision in PR #1622 before deploying— @JustAlex3125 if you’re able to turn this one around really quickly (verifying of course that there are no regression) then that would be great!

**Comment by dondi** [comment]
Beta has been updated to 7.9.6, including PRs #1605 (#1460, #1602), #1618 (#1417, #1614), #1621 (#1613), and #1622 (#1393), with corresponding issues addressed in parentheses respectively

**Comment by dondi** [comment]
Beta has been updated to 7.9.7, including PR #1624 for issue #1617

**Comment by dondi** [comment]
Beta has been updated to 7.9.8, including PR #1625 and #1626, representing follow-up formatting for #1617 and addressing #1611, respectively

**Comment by dondi** [comment]
Beta has been updated to 7.9.9, including PR #1628 which seeks to address issue #1610

**Comment by dondi** [comment]
Beta has been updated to 7.9.10, including PRs #1631, #1633, #1634, and #1635, which seek to address issues #1357, another discovered server-crashing file, #1460, and #1623, respectively

**Comment by dondi** [comment]
Beta has been updated to 7.9.11, including PR #1636, which seeks to address issue #1611

**Comment by dondi** [comment]
Well here’s an atypical update: @jiacelyne’s PR #1638, implementing the change from `||` to `??` in accordance with newer versions of JavaScript, turned out to be…_too new_ for the version of JavaScript running on the GRNsight server 😮 As such, I reverted the commit and ultimately, there is no update to beta

Just noting here that although `??` is the way to go, we will probably not be able to take advantage of that until we switch out of classic (or until the software stack on the server can be updated further)

**Comment by dondi** [comment]
Beta has been updated to 7.9.12, include PRs #1641 and #1644, which seek to address multiple two-column warning issues and #1357, respectively

**Comment by dondi** [comment]
Beta has been updated to 7.9.13 with PR #1650, addressing issue #1293
